### PR TITLE
ZEN-32163: CZ is not sending component group membership as metadata

### PR DESCRIPTION
**Pull request by roxy-zenoss**

### CODE REVIEW

**Comment by webhookbotuser** [comment]
Can one of the admins verify this patch?

**Comment by webhookbotuser** [comment]
Can one of the admins verify this patch?

**Comment by webhookbotuser** [comment]
Can one of the admins verify this patch?

**Comment by webhookbotuser** [comment]
Can one of the admins verify this patch?

**Comment by webhookbotuser** [comment]
Can one of the admins verify this patch?

**Comment by Deer-WarLord** [comment]
retest this

**Comment by roxy-zenoss** [comment]
retest this

File: Products/Zing/datamaps.py [hunk 1]
```diff
@@ -153,10 +153,14 @@ def _generate_facts(self, facts_per_device, zing_tx_state):
                 # organizers and impact relationships facts for the component
                 comp_uuid = f.metadata.get(ZFact.FactKeys.CONTEXT_UUID_KEY, "")
                 if comp_uuid:
+                    comp_groups = []
+                    for component in device.getDeviceComponents():
+                        if component.getUUID() == comp_uuid:
+                            comp_groups = component.getComponentGroupNames()
```
Comment (jpeacock-zenoss) [inline Products/Zing/datamaps.py]: After retrieving the component group names, there should be a `break` here to stop the `for` loop.  There's no point to continue looping, is there?

**Comment by jpeacock-zenoss** [comment]
retest this

**Pull request merged**